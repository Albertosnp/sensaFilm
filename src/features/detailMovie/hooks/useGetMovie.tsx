import { useEffect, useState } from 'react';
import { MovieDetailI } from '@/types';
import { getMovie } from '@/features/detailMovie/api/getMovie';

interface UseGetMovieProps {
  idMovie: string | undefined;
}

export const useGetMovie = ({ idMovie }: UseGetMovieProps) => {
  const [movie, setMovie] = useState<MovieDetailI[] | null>(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    if (!idMovie) return;

    let cancelled = true;
    setLoading(true);

    getMovie(idMovie)
      .then((res) => {
        if (!cancelled) return;

        setMovie(movie);
        setLoading(false);
      })
      .catch((err) => console.error(err));

    return () => {
      cancelled = false;
    };
  }, [idMovie]);

  return { movie, loading } as const;
};
