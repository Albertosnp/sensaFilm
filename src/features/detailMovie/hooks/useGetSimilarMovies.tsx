import { useEffect, useState } from 'react';
import { MovieI } from '@/types';
import { getSimilarMovies } from '../api/getSimilarMovies';

interface UseGetMovieProps {
  idMovie: string | undefined;
}

export const useGetSimilarMovies = ({ idMovie }: UseGetMovieProps) => {
  const [movies, setMovies] = useState<MovieI[] | null>(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    if (!idMovie) return;

    let cancelled = true;
    setLoading(true);

    getSimilarMovies(idMovie)
      .then((res) => {
        if (!cancelled) return;

        const similarMovies = res?.results;
        setMovies(similarMovies);
        setLoading(false);
      })
      .catch((err) => console.error(err));

    return () => {
      cancelled = false;
    };
  }, [idMovie]);

  return { movies, loading } as const;
};
