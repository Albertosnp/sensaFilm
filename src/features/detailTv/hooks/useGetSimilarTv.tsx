import { useEffect, useState } from 'react';
import { TvProgramI } from '@/types';
import { getSimilarTv } from '../api/getSimilarTv';

interface UseGetMovieProps {
  idTv: string | undefined;
}

export const useGetSimilarTv = ({ idTv }: UseGetMovieProps) => {
  const [tvPrograms, setTvPrograms] = useState<TvProgramI[] | null>(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    if (!idTv) return;

    let cancelled = true;
    setLoading(true);

    getSimilarTv(idTv)
      .then((res) => {
        if (!cancelled) return;

        const programs = res?.results;

        setTvPrograms(programs);
        setLoading(false);
      })
      .catch((err) => console.error(err));

    return () => {
      cancelled = false;
    };
  }, [idTv]);

  return { tvPrograms, loading } as const;
};
