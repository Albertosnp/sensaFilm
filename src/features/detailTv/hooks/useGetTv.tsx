import { TvDetailI } from '@/types';
import { useEffect, useState } from 'react';
import { getTvProgram } from '../api/getTvProgram';

interface UseGetMovieProps {
  idTv: string | undefined;
}

export const useGetTv = ({ idTv }: UseGetMovieProps) => {
  const [tvDetail, setTvDetail] = useState<TvDetailI[] | null>(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    if (!idTv) return;

    let cancelled = true;
    setLoading(true);

    getTvProgram(idTv)
      .then((res) => {
        if (!cancelled) return;

        setTvDetail(tvDetail);
        setLoading(false);
      })
      .catch((err) => console.error(err));

    return () => {
      cancelled = false;
    };
  }, [idTv]);

  return { tvDetail, loading } as const;
};
