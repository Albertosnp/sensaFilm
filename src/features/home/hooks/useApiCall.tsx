import { useEffect, useState } from 'react';

export const useApiCall = () => {
  const [result, setResult] = useState();
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    let cancelled = true;
    setLoading(true);

    //API call

    return () => {
      cancelled = false;
    };
  }, []);

  return [result, loading] as const;
};
