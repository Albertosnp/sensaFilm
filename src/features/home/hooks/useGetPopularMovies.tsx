import { useEffect, useState } from 'react';
import { getPopularMovies } from '../api/getPopularMovies';
import { MovieI } from '@/types';

export const useGetPopularMovies = () => {
  const [popularMovies, setPopularMovies] = useState<MovieI[] | null>(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    let cancelled = true;
    setLoading(true);

    getPopularMovies()
      .then((res) => {
        if (!cancelled) return;

        const movies = res?.results;
        setTimeout(() => {
          setPopularMovies(movies);

          setLoading(false);
        }, 2000);
      })
      .catch((err) => console.error(err));

    return () => {
      cancelled = false;
    };
  }, []);

  return { popularMovies, loading } as const;
};
