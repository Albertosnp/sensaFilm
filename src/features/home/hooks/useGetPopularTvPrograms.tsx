import { useEffect, useState } from 'react';
import { getPopularTvPrograms } from '../api/getPopularTvPrograms';
import { TvProgramI } from '@/types';

export const useGetPopularTvPrograms = () => {
  const [popularTvPrograms, setPopularTvPrograms] = useState<
    TvProgramI[] | null
  >(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    let cancelled = true;
    setLoading(true);

    getPopularTvPrograms()
      .then((res) => {
        if (!cancelled) return;

        const tvPrograms = res?.results;

        setPopularTvPrograms(tvPrograms);
        setLoading(false);
      })
      .catch((err) => console.error(err));

    return () => {
      cancelled = false;
    };
  }, []);

  return { popularTvPrograms, loading } as const;
};
