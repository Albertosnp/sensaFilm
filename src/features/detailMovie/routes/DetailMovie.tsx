import { useParams } from 'react-router-dom';
import { SwiperSlide } from 'swiper/react';
import { Carousel, DetailLayout, GenericCard } from '@/components';
import { useGetMovie } from '@/features/detailMovie/hooks/useGetMovie';
import { useGetSimilarMovies } from '../hooks/useGetSimilarMovies';

export const DetailMovie = () => {
  const { idMovie } = useParams();
  const { movie } = useGetMovie({ idMovie });
  const { movies } = useGetSimilarMovies({ idMovie });

  return (
    <main>
      {movie && (
        <DetailLayout {...movie}>
          {movies && movies.length > 0 && (
            <Carousel>
              {movies?.map(({ id, poster_path, title, vote_average }) => (
                <SwiperSlide key={`${id}-movie`}>
                  <GenericCard
                    key={`${id}-movie`}
                    id={id}
                    title={title}
                    image={poster_path ?? ''}
                    vote_average={vote_average}
                    path="/detailMovie"
                  />
                </SwiperSlide>
              ))}
            </Carousel>
          )}
        </DetailLayout>
      )}
    </main>
  );
};
