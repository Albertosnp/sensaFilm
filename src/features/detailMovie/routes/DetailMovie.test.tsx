import { createRoot } from "react-dom/client";
import { useParams } from "react-router-dom";
import { DetailMovie } from "./DetailMovie";
import { render } from "@testing-library/react";
import { vi } from "vitest";
import { useGetMovie } from "../hooks/useGetMovie";
import { useGetSimilarMovies } from "../hooks/useGetSimilarMovies";
import React from "react";

vi.mock("react-router-dom", async () => {
  return {
    ...vi.importMock("react-router-dom"),
    useParams: vi.fn(),
    Link: ({ children, to }: { children: JSX.Element; to: string }) =>
      React.createElement("a", { href: to }, children),
  };
});

vi.mock("../hooks/useGetMovie", () => {
  return {
    ...vi.importMock("../hooks/useGetMovie"),
    useGetMovie: vi.fn(),
  };
});

vi.mock("../hooks/useGetSimilarMovies", () => {
  return {
    ...vi.importMock("../hooks/useGetSimilarMovies"),
    useGetSimilarMovies: vi.fn(),
  };
});

const movies = [
  {
    adult: false,
    backdrop_path: null,
    genre_ids: [99],
    id: 1158004,
    original_language: "en",
    original_title: "Careless Reef: Part 2: Abu Kiffan",
    overview: "",
    popularity: 0.6,
    poster_path: null,
    release_date: "2005-07-01",
    title: "Careless Reef: Part 2: Abu Kiffan",
    video: false,
    vote_average: 0,
    vote_count: 0,
  },
  {
    adult: false,
    backdrop_path: null,
    genre_ids: [16, 10402],
    id: 1148993,
    original_language: "en",
    original_title: "Bug Fixed",
    overview: "",
    popularity: 0.6,
    poster_path: "/xzQS1AeZAuhJZdQa5bko0GDnAsx.jpg",
    release_date: "1994-01-12",
    title: "Bug Fixed",
    video: false,
    vote_average: 0,
    vote_count: 0,
  },
  {
    adult: false,
    backdrop_path: "/qGwEyTNBoGQ0e5XBYNUdAPS292T.jpg",
    genre_ids: [9648, 53, 18],
    id: 1146585,
    original_language: "ar",
    original_title: "منتصف الليل",
    overview: "",
    popularity: 1.113,
    poster_path: "/qzIi2wjm3mMWm4GNJzolF9E3zuR.jpg",
    release_date: "2023-07-05",
    title: "منتصف الليل",
    video: false,
    vote_average: 10,
    vote_count: 1,
  },
];
const movie = {
  adult: false,
  backdrop_path: "/jZIYaISP3GBSrVOPfrp98AMa8Ng.jpg",
  belongs_to_collection: null,
  budget: 200000000,
  genres: [
    {
      id: 16,
      name: "Animación",
    },
    {
      id: 35,
      name: "Comedia",
    },
    {
      id: 10751,
      name: "Familia",
    },
    {
      id: 14,
      name: "Fantasía",
    },
    {
      id: 10749,
      name: "Romance",
    },
  ],
  homepage: "https://www.disney.es/peliculas/elemental",
  id: 976573,
  imdb_id: "tt15789038",
  original_language: "en",
  original_title: "Elemental",
  overview:
    "En una ciudad donde los residentes del fuego, el agua, la tierra y el aire viven juntos, una joven apasionada y un chico que se deja llevar por la corriente descubrirán algo elemental: cuánto tienen en común.",
  popularity: 3167.104,
  poster_path: "/6oH378KUfCEitzJkm07r97L0RsZ.jpg",
  production_companies: [
    {
      id: 2,
      logo_path: "/wdrCwmRnLFJhEoH8GSfymY85KHT.png",
      name: "Walt Disney Pictures",
      origin_country: "US",
    },
    {
      id: 3,
      logo_path: "/1TjvGVDMYsj6JBxOAkUHpPEwLf7.png",
      name: "Pixar",
      origin_country: "US",
    },
  ],
  production_countries: [
    {
      iso_3166_1: "US",
      name: "United States of America",
    },
  ],
  release_date: "2023-06-14",
  revenue: 458822015,
  runtime: 102,
  spoken_languages: [
    {
      english_name: "Spanish",
      iso_639_1: "es",
      name: "Español",
    },
    {
      english_name: "English",
      iso_639_1: "en",
      name: "English",
    },
  ],
  status: "Released",
  tagline: "Los opuestos se atraen.",
  title: "Elemental",
  video: false,
  vote_average: 7.787,
  vote_count: 1474,
};

describe("DetailMovie", () => {
  const defaultProps = {};
  const getComponent = (props = defaultProps) => <DetailMovie {...props} />;
  const setup = (props = defaultProps) => render(getComponent(props));

  beforeEach(() => {
    // @ts-ignore
    useParams.mockReturnValue({ idMovie: "976573" });
    // @ts-ignore
    useGetMovie.mockReturnValue({ movie });
    // @ts-ignore
    useGetSimilarMovies.mockReturnValue({ movies });
  });

  it("Should render without crashing", () => {
    const div = document.createElement("div");
    createRoot(div).render(getComponent());
    createRoot(div).unmount();
  });

  it("Should match snapshot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
