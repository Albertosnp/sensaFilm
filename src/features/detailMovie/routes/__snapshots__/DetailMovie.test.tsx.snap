// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DetailMovie > Should match snapshot 1`] = `
<div>
  <main>
    <section
      class="DetailLayout css-x3l6mn"
    >
      <div
        class="DetailLayout__gradient"
      />
      <article
        class="DetailLayout__info"
      >
        <img
          class="DetailLayout__info__imgPoster"
          src="https://image.tmdb.org/t/p/w500//6oH378KUfCEitzJkm07r97L0RsZ.jpg"
        />
        <div
          class="DetailLayout__info__detailsBlock"
        >
          <h3
            class="DetailLayout__info__detailsBlock__title"
          >
            Elemental
          </h3>
          <h6
            class="DetailLayout__detailTitle"
          >
            Sinopsis:
          </h6>
          <p
            class="DetailLayout__info__detailsBlock__overview"
          >
            En una ciudad donde los residentes del fuego, el agua, la tierra y el aire viven juntos, una joven apasionada y un chico que se deja llevar por la corriente descubrirán algo elemental: cuánto tienen en común.
          </p>
        </div>
      </article>
      <div
        class="Carousel css-ftttuy"
      >
        <div
          class="swiper swiper-initialized swiper-horizontal slider"
        >
          <div
            class="swiper-wrapper"
            style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
          >
            <div
              class="swiper-slide swiper-slide-duplicate"
              data-swiper-slide-index="0"
            >
              <figure
                class="GenericCard css-1sbhet6"
              >
                <a
                  href="/detailMovie/1158004"
                >
                  <img
                    alt="imagen-picture"
                    class="GenericCard__image"
                    src="https://image.tmdb.org/t/p/w500/"
                  />
                  <div
                    class="GenericCard__footer"
                  >
                    <p
                      class="GenericCard__footer__title"
                    >
                      Careless Reef: Part 2: Abu Kiffan
                    </p>
                    <span
                      class="GenericCard__footer__votes"
                    >
                      0
                      /10
                    </span>
                  </div>
                </a>
              </figure>
            </div>
            <div
              class="swiper-slide swiper-slide-duplicate"
              data-swiper-slide-index="1"
            >
              <figure
                class="GenericCard css-1sbhet6"
              >
                <a
                  href="/detailMovie/1148993"
                >
                  <img
                    alt="imagen-picture"
                    class="GenericCard__image"
                    src="https://image.tmdb.org/t/p/w500//xzQS1AeZAuhJZdQa5bko0GDnAsx.jpg"
                  />
                  <div
                    class="GenericCard__footer"
                  >
                    <p
                      class="GenericCard__footer__title"
                    >
                      Bug Fixed
                    </p>
                    <span
                      class="GenericCard__footer__votes"
                    >
                      0
                      /10
                    </span>
                  </div>
                </a>
              </figure>
            </div>
            <div
              class="swiper-slide swiper-slide-duplicate"
              data-swiper-slide-index="2"
            >
              <figure
                class="GenericCard css-1sbhet6"
              >
                <a
                  href="/detailMovie/1146585"
                >
                  <img
                    alt="imagen-picture"
                    class="GenericCard__image"
                    src="https://image.tmdb.org/t/p/w500//qzIi2wjm3mMWm4GNJzolF9E3zuR.jpg"
                  />
                  <div
                    class="GenericCard__footer"
                  >
                    <p
                      class="GenericCard__footer__title"
                    >
                      منتصف الليل
                    </p>
                    <span
                      class="GenericCard__footer__votes"
                    >
                      10
                      /10
                    </span>
                  </div>
                </a>
              </figure>
            </div>
            <div
              class="swiper-slide"
              data-swiper-slide-index="0"
            >
              <figure
                class="GenericCard css-1sbhet6"
              >
                <a
                  href="/detailMovie/1158004"
                >
                  <img
                    alt="imagen-picture"
                    class="GenericCard__image"
                    src="https://image.tmdb.org/t/p/w500/"
                  />
                  <div
                    class="GenericCard__footer"
                  >
                    <p
                      class="GenericCard__footer__title"
                    >
                      Careless Reef: Part 2: Abu Kiffan
                    </p>
                    <span
                      class="GenericCard__footer__votes"
                    >
                      0
                      /10
                    </span>
                  </div>
                </a>
              </figure>
            </div>
            <div
              class="swiper-slide"
              data-swiper-slide-index="1"
            >
              <figure
                class="GenericCard css-1sbhet6"
              >
                <a
                  href="/detailMovie/1148993"
                >
                  <img
                    alt="imagen-picture"
                    class="GenericCard__image"
                    src="https://image.tmdb.org/t/p/w500//xzQS1AeZAuhJZdQa5bko0GDnAsx.jpg"
                  />
                  <div
                    class="GenericCard__footer"
                  >
                    <p
                      class="GenericCard__footer__title"
                    >
                      Bug Fixed
                    </p>
                    <span
                      class="GenericCard__footer__votes"
                    >
                      0
                      /10
                    </span>
                  </div>
                </a>
              </figure>
            </div>
            <div
              class="swiper-slide"
              data-swiper-slide-index="2"
            >
              <figure
                class="GenericCard css-1sbhet6"
              >
                <a
                  href="/detailMovie/1146585"
                >
                  <img
                    alt="imagen-picture"
                    class="GenericCard__image"
                    src="https://image.tmdb.org/t/p/w500//qzIi2wjm3mMWm4GNJzolF9E3zuR.jpg"
                  />
                  <div
                    class="GenericCard__footer"
                  >
                    <p
                      class="GenericCard__footer__title"
                    >
                      منتصف الليل
                    </p>
                    <span
                      class="GenericCard__footer__votes"
                    >
                      10
                      /10
                    </span>
                  </div>
                </a>
              </figure>
            </div>
            <div
              class="swiper-slide swiper-slide-duplicate"
              data-swiper-slide-index="0"
            >
              <figure
                class="GenericCard css-1sbhet6"
              >
                <a
                  href="/detailMovie/1158004"
                >
                  <img
                    alt="imagen-picture"
                    class="GenericCard__image"
                    src="https://image.tmdb.org/t/p/w500/"
                  />
                  <div
                    class="GenericCard__footer"
                  >
                    <p
                      class="GenericCard__footer__title"
                    >
                      Careless Reef: Part 2: Abu Kiffan
                    </p>
                    <span
                      class="GenericCard__footer__votes"
                    >
                      0
                      /10
                    </span>
                  </div>
                </a>
              </figure>
            </div>
            <div
              class="swiper-slide swiper-slide-duplicate"
              data-swiper-slide-index="1"
            >
              <figure
                class="GenericCard css-1sbhet6"
              >
                <a
                  href="/detailMovie/1148993"
                >
                  <img
                    alt="imagen-picture"
                    class="GenericCard__image"
                    src="https://image.tmdb.org/t/p/w500//xzQS1AeZAuhJZdQa5bko0GDnAsx.jpg"
                  />
                  <div
                    class="GenericCard__footer"
                  >
                    <p
                      class="GenericCard__footer__title"
                    >
                      Bug Fixed
                    </p>
                    <span
                      class="GenericCard__footer__votes"
                    >
                      0
                      /10
                    </span>
                  </div>
                </a>
              </figure>
            </div>
            <div
              class="swiper-slide swiper-slide-duplicate"
              data-swiper-slide-index="2"
            >
              <figure
                class="GenericCard css-1sbhet6"
              >
                <a
                  href="/detailMovie/1146585"
                >
                  <img
                    alt="imagen-picture"
                    class="GenericCard__image"
                    src="https://image.tmdb.org/t/p/w500//qzIi2wjm3mMWm4GNJzolF9E3zuR.jpg"
                  />
                  <div
                    class="GenericCard__footer"
                  >
                    <p
                      class="GenericCard__footer__title"
                    >
                      منتصف الليل
                    </p>
                    <span
                      class="GenericCard__footer__votes"
                    >
                      10
                      /10
                    </span>
                  </div>
                </a>
              </figure>
            </div>
          </div>
        </div>
      </div>
    </section>
  </main>
</div>
`;
